app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -38,13 +48,30 @@ public void done(@Nullable JSONObject jsonObject) {
                 try {
 
                     JSONArray groceryJsonArray = jsonObject.optJSONArray(JsonKeys.RESULTS);
-
-                    for (int i = 0; i < groceryJsonArray.length(); i++) {
-                        RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
-                        groceryItem.setName(groceryJsonArray.getJSONObject(i).optString(JsonKeys.NAME));
+                    if (groceryJsonArray != null) {
+                        for (int i = 0; i < groceryJsonArray.length(); i++) {
+                            RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
+                            groceryItem.setGroceryId(groceryJsonArray.getJSONObject(i)
+                                    .getString(JsonKeys.OBJECTID));
+                            groceryItem.setName(groceryJsonArray.getJSONObject(i)
+                                    .getString(JsonKeys.NAME));
+                            groceryItem.setGroupId(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.GROUPID).getString(JsonKeys.OBJECTID));
+                            groceryItem.setGroupName(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.GROUPID).getString(JsonKeys.NAME));
+                            groceryItem.setCreatedBy(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.CREATEDBY).getString(JsonKeys.USERNAME));
+                            JSONObject purchasedByObj = groceryJsonArray.getJSONObject(i)
+                                    .optJSONObject(JsonKeys.PURCHASEDBY);
+                            if (purchasedByObj != null) {
+                                groceryItem.setPurchasedBy(purchasedByObj.getString(JsonKeys.USERNAME));
+                            }
+                        }
+
+                        realm.commitTransaction();
```

Comment:
If you cancel your transaction, you should call promise.onFailure(); This code will call promise.onSuccess().